src/Protocol/Document/TextDocumentAttributes.cs

```diff
@@ -2,8 +2,10 @@
 
 namespace OmniSharp.Extensions.LanguageServer.Protocol.Document
 {
-    public class TextDocumentAttributes : IEquatable<TextDocumentAttributes>
+    public record TextDocumentAttributes
     {
+        public TextDocumentAttributes() { }
```

Comment:
Is there a particular reason we need the empty constructor?

Reply:
Backwards compatibility

Reply:
oh and json serialization I found out later, lol.

Reply:
Ah, yeah I've run into the json thing before, it's a real pain.